### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by groodt**
Issue awslabs#139

Description of changes:
Clarify documentation around Spark version support.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,8 +8,10 @@ Deequ is a library built on top of Apache Spark for defining "unit tests for dat
 
 ## Requirements and Installation
 
-__Deequ__ depends on Java 8 and Apache Spark 2.2, and is available via [maven central](http://mvnrepository.com/artifact/com.amazon.deequ/deequ). Add the latest release as a dependency to your project:
+__Deequ__ depends on Java 8 and Apache Spark 2.2. It only uses non-experimental APIs, which should work up to Spark 2.4.
```
Comment (sscdotopen) [inline README.md]: It is true that we support Spark 2.3 and 2.4.

However, we extend some internal catalyst-based aggregation functions and have to apply some hacks to get this to work across different versions: https://github.com/awslabs/deequ/blob/master/src/main/scala/com/amazon/deequ/analyzers/catalyst/AttributeReferenceCreation.java#L73

So the comment about the non-experimental APIs is not true unfortunately.

Comment (groodt) [reply README.md]: Ok, thanks for clarifying. I've updated the wording a bit to make the risks clearer.

File: README.md [hunk 2]
```diff
@@ -8,8 +8,10 @@ Deequ is a library built on top of Apache Spark for defining "unit tests for dat
 
 ## Requirements and Installation
 
-__Deequ__ depends on Java 8 and Apache Spark 2.2, and is available via [maven central](http://mvnrepository.com/artifact/com.amazon.deequ/deequ). Add the latest release as a dependency to your project:
+__Deequ__ depends on Java 8 and is known to work with Apache Spark versions 2.2.x to 2.4.x. However, be aware of the risk that deequ is extending certain internal Spark APIs which are subject to change in future and may break as Spark versions change.
```
Comment (sscdotopen) [inline README.md]: I think the "However ..." sentence unnecessarily scares users. Maybe we should just say that it works up to 2.4

Comment (groodt) [reply README.md]: I think that's fine too. Ok, I'll remove that entire sentence. 👍

**Comment by sscdotopen** [comment]
Thank you for your contribution!

**Pull request merged**